src/bionics_ui.cpp

```diff
@@ -246,144 +278,113 @@ static void draw_connectors( const catacurses::window &win, const int start_y, c
+            if( BionicsDisplayType::displayTypes[i].ident() == dType ) {
+                bionics_by_type[i].emplace_back( &b );
+                if( bionics_by_type[i].size() > max_list_size ) {
+                    max_list_size = bionics_by_type[i].size();
+                }
+                break;
             }
         }
     }
-    return type;
-}
+    const int SLOTS_WIDTH = 29;
 
-static std::vector<bionic *> filtered_bionics( bionic_collection &all_bionics,
-        bionic_tab_mode mode )
-{
-    std::vector< bionic *>filtered_entries;
-    for( auto &elem : all_bionics ) {
-        if( ( mode == TAB_ACTIVE ) == elem.id->activated ) {
-            filtered_entries.push_back( &elem );
-        }
+    max_name_length += 2; //invlet
+    size_t col_width = with_slots ? SLOTS_WIDTH : 0;
+    for( size_t i = 0; i < column_num_entries; i++ ) {
+        col_width += get_col_data( i ).get_width();
     }
-    return filtered_entries;
-}
 
-void player::power_bionics()
-{
-    std::vector <bionic *> passive = filtered_bionics( *my_bionics, TAB_PASSIVE );
-    std::vector <bionic *> active = filtered_bionics( *my_bionics, TAB_ACTIVE );
-    bionic *bio_last = nullptr;
-    bionic_tab_mode tab_mode = TAB_ACTIVE;
+    size_t tabs_width = 1;
+    for( size_t i = 0; i < bionics_by_type.size(); i++ ) {
+        if( !bionics_by_type.empty() ) {
+            tabs_width += utf8_width( BionicsDisplayType::displayTypes[i].display_string() ) + 3;
+        }
+    }
 
     //added title_tab_height for the tabbed bionic display
-    int TITLE_HEIGHT = 2;
-    int TITLE_TAB_HEIGHT = 3;
+    const int TITLE_HEIGHT = 1;
+    const int TITLE_TAB_HEIGHT = 3;
+    const int FOOTER_HEIGHT = 5;
 
     // Main window
-    /** Total required height is:
-     * top frame line:                                         + 1
-     * height of title window:                                 + TITLE_HEIGHT
-     * height of tabs:                                         + TITLE_TAB_HEIGHT
-     * height of the biggest list of active/passive bionics:   + bionic_count
-     * bottom frame line:                                      + 1
-     * TOTAL: TITLE_HEIGHT + TITLE_TAB_HEIGHT + bionic_count + 2
-     */
-    const int HEIGHT = std::min( TERMY,
-                                 std::max( FULL_SCREEN_HEIGHT,
-                                           TITLE_HEIGHT + TITLE_TAB_HEIGHT +
-                                           static_cast<int>( my_bionics->size() ) + 2 ) );
-    const int WIDTH = FULL_SCREEN_WIDTH + ( TERMX - FULL_SCREEN_WIDTH ) / 2;
-    const int START_X = ( TERMX - WIDTH ) / 2;
-    const int START_Y = ( TERMY - HEIGHT ) / 2;
-    //wBio is the entire bionic window
-    catacurses::window wBio = catacurses::newwin( HEIGHT, WIDTH, point( START_X, START_Y ) );
+    // +3 = top + bottom + footer line
+    const int requested_height = TITLE_HEIGHT + TITLE_TAB_HEIGHT + FOOTER_HEIGHT + max_list_size + 3;
```

Comment:
You have one unaccounted line.
```suggestion
    const int requested_height = TITLE_HEIGHT + TITLE_TAB_HEIGHT + FOOTER_HEIGHT + max_list_size + 4;
```

Reply:
Thanks for testing